src/modules/kvs/commit.c

```diff
@@ -242,24 +242,34 @@ static int store_cache (commit_t *c, int current_epoch, json_t *o,
  */
 static int commit_unroll (commit_t *c, int current_epoch, json_t *dir)
 {
-    json_t *value;
-    json_t *subdir, *key_value;
-    json_t *tmpdirent;
+    json_t *dir_entry;
+    json_t *dir_data;
+    json_t *tmp;
     href_t ref;
     int ret;
     struct cache_entry *hp;
-    void *iter = json_object_iter (dir);
+    void *iter;
+
+    if (!treeobj_is_dir (dir)) {
+        errno = EPERM;
```

Comment:
Yeah, this one in ```commit_unroll()``` is specifically an "assert"-like check (a check earlier in ```commit_link_dirent()``` would return an error to the user if they passed in a non-dir root).  Makes sense that we can convert it to assert outright.